File: old/sorting/quick_sort/implement_lomuto_partition/implement_lomuto_partition.py
def lomuto_quick_sort_index(arr, start, end):
    left = start
    pivot = arr[end]

    for i in range(start, end):
        if arr[i] < pivot:
            arr[i], arr[left] = arr[left], arr[i]
            left += 1

    arr[end] = arr[left]
    arr[left] = pivot

    return left


def main():
    input = [3, 2, 1, 5, 4]
    output = lomuto_quick_sort_index(input, 0, 4)
    print(output)


main()

File: old/sorting/quick_sort/implement_lomuto_partition/implement_lomuto_partition.cpp
#include <iostream>
#include <vector>

using namespace std;

int lomuto_partition_index(vector<int>& arr, int start, int end) {
    int pivot = arr[end];
    int left = start;

    for (int i = start; i < end; i++) {
        if (arr[i] < pivot) {
            swap(arr[i], arr[left]);
            left++;
        }
    }

    arr[end] = arr[left];
    arr[left] = pivot;

    return left;
}

int main() {
    vector<int> input = {3, 2, 1, 5, 4};
    int output = lomuto_partition_index(input, 0, 4);
    cout << "index: " << output;
    return 0;
}
